temp

DrewKimball opened the pull request.

DrewKimball committed:
opt: refactor join ordering to allow pruning

This commit refactors the join re-ordering logic to allow for some
candidate plans to be pruned away before they are built into the memo.
Note that this commit does not add any pruning logic.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/129397)
<!-- Reviewable:end -->

DrewKimball committed:
opt: use cardinality upper bounds to prune join orders

This commit adds logic to compute upper bounds on the cardinality of
a join before it is built, using max value frequency statistics for the
columns used in the join conditions. The upper bounds are used in a simple
cost function `C_out`, which simply sums input and output cardinalities.
Join orders that are >10x worse than the best cost are not considered.

Epic: None
Release note: None

DrewKimball committed:
opt: remove join order pruning

This commit comments out the code snippet that removes join plans with
too high an estimated cost, so that pruned plans are displayed in the tests.

Epic: None
Release note: None
